File: pkg/bot_utils/Maintenance.go
package bot_utils

import "main/pkg/logger"

var isMaintenanceMode bool = false

func SetMaintenanceMode(mode bool) {
	isMaintenanceMode = mode
	logger.WarningLogger.Println("Maintenance mode set to:", mode)
}

func IsMaintenanceMode() bool {
	return isMaintenanceMode
}

File: utils/maintenance.ts
let maintenanceMode = false;

export function setMaintenanceMode(enabled: boolean): void {
    maintenanceMode = enabled;
}

export function isMaintenanceMode(): boolean {
    return maintenanceMode;
}

export function toggleMaintenanceMode(): boolean {
    maintenanceMode = !maintenanceMode;
    return maintenanceMode;
}
